bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiClient.java

```diff
@@ -1004,16 +1005,24 @@ protected Shelly2RpcBaseMessage buildRequest(String method, @Nullable Object par
         return request;
     }
 
-    protected String mapValue(Map<String, String> map, @Nullable String key) {
-        if (key == null || key.isEmpty()) {
+    protected String mapValue(Map<String, String> map, String key) {
+        String value = map.getOrDefault(key, "");
+        if (value.isEmpty()) {
```

Comment:
This would be more straight-forward, since I assume that the provided `Map`s don't contain empty strings as values.
```suggestion
        String value = map.get(key);
        if (value == null) {
```